app/routes/public/sessions.js

```diff
@@ -140,7 +141,44 @@ export default class SessionsRoute extends Route {
         perPageParam : 'page[size]',
         pageParam    : 'page[number]',
         store        : eventDetails
-      })
-    };
+      }),
+      dates: this.getDates(eventDetails)
+    });
+  }
+
+  async getDates(event) {
+
+    const scheduledFilterOptions = [
+      {
+        and: [
+          {
+            name : 'starts-at',
+            op   : 'ne',
+            val  : null
+          },
+          {
+            or: [
+              {
+                name : 'state',
+                op   : 'eq',
+                val  : 'accepted'
+              },
+              {
+                name : 'state',
+                op   : 'eq',
+                val  : 'confirmed'
+              }
+            ]
+          }
+        ]
+      }
+    ];
+
+    const sessions = await event.query('sessions', {
+      filter       : scheduledFilterOptions,
+      'page[size]' : 0
```

Comment:
I am so confused. Why did you remove fields[session]=starts-at from here

Did you not understand why I added it?

Reply:
yes sir i got that , in order to get the starting dates of all sessions  , but i checked after removing that , it was working just fine , that's why i removed , i am adding it again.

Reply:
LOL, it was not added just for the sake of it, now it is loading 2 MB of data from server instead of 256KB

Reply:
yes sir I didn't understood it properly, I was not able to solve the issue,  sorry for that,  I will not bother you again.